src/commands/commandRegistry.ts

```diff
@@ -31,21 +33,25 @@ export class CommandRegistry {
       this._commands.set(phrase, existingCommand as ICommand);
     } else {
       // create new instance of the command and map it to the phrase
-      this._commands.set(phrase, new (command as any)());
-      command.onload?.();
+      const instance = new (command as any)();
+      this._commands.set(phrase, instance);
+      instance.onLoad?.();
     }
   }
 
   execute(...args) {
-    // Root command
+    // Root command instance
     const command = this._commands.get(args[0]);
     // Phrase after root command
     const subCommandPhrase = args[1][0];
     // if the 2nd phrase matches a subcommand then use it. Otherwise default to the default method for the command class.
     if (command['commands'].get(subCommandPhrase)) {
       command[command['commands'].get(subCommandPhrase)](args[1].splice(1), args[2]);
     } else {
-      command[command['commands'].get(undefined)](args[1], args[2]);
+      // gets the function where @Invoke() was parameterless
+      const func = command[command['commands'].get(undefined)];
+      // apply the command instance to the function call to perserve the 'this' context
+      func.apply(command, [args[1], args[2]]);
```

Comment:
First major issue.
The old function call was calling the methods on the class prototype instead of the class instance. This caused the `this` context to be undefined inside the command. Now we apply the correct command instance to the function call to have the correct `this` context